sonoshell/Dockerfile

```diff
@@ -0,0 +1,20 @@
+FROM golang:1.17-buster as build
+
+run mkdir -p /_sonoshell
+COPY ./pkg /_sonoshell/pkg
+COPY ./go.sum /_sonoshell/go.sum
+COPY ./go.mod /_sonoshell/go.mod
+
+WORKDIR /_sonoshell
+
+# Needed to build successfully
+RUN go mod tidy
+
+RUN go build -o sonoshell ./pkg
+
+FROM debian:buster-slim
+
+COPY --from=build /_sonoshell/sonoshell /sonoshell
+COPY ./test.yaml /test.yaml
```

Comment:
I'll update this to create a configmap, but I have one question about testing; is there a preferred way that I could create a test.yaml file in the image so that it has something to run against when I build it locally? Currently the workflow that I use to verify changes is to run 
```shell
docker build -t ${image_name} .
docker run ${image_name}
```

As it stands, this wouldn't load a configmap from a `plugin.yaml` type file; would it be preferable to leave the baked-in .yaml file for testing, create it in the image, or something else?

Reply:
I see, so the question is how best to test this without running it in K8s/sonobuoy at all?

In that case you just need to mount the script manually for docker:
```
docker build -t ${image_name} .
docker run 
  -v $(pwd)/test.yaml:/manualMount/test.yaml \
  laevos/sonoshell:latest \
  ./sonoshell /manualMount/test.yaml
```

Of course, if you want to hardcode one random text file as an example/test for itself it wouldn't be the end of the world. But I wouldn't recommend users change that as the flow for their own work. It would just be a built-in test effectively that users wouldn't ever want to run themselves. I would make the sample script be very explicit (e.g. have a test with names like 'this is a sample script; yours should be provided via...')

I think always keeping the script out is better though and would make it easier for you to have N different scripts you may target for tests and you just specify which to run. And if you expand this to allow running N test files in series/parallel then you avoid adding more and more unnecessary script files into the core image.

Reply:
For now made a change to make explicit in the renamed `.test.yaml` that this is solely for testing the docker image locally, and that actual tests should be written as a config-map. Are there any other changes we'd want to allow the simultaneous running of multiple test suites or anything like that, or should that be handled already by utilizing the existing config-map and having tests run in series/parallel per the `command` portion of the plugin yaml?

Reply:
You have it read just the first argument and process that file. It wouldn't be hard to extend this to just loop over those arguments and operate on each of them.

However, I'd rather get this in as-is then extend the functionality if you find that you want/need it.

The configmap mechanism can hold multiple files so you could automatically load every file there (and modify the code so that if a directory is given as an argument all files in that directory are processed). Again, I think thats an extension if we want it next.